extensions/notebook/src/jupyter/serverInstance.ts

```diff
@@ -168,9 +186,11 @@ export class PerNotebookServerInstance implements IServerInstance {
 			// For now, we don't care as this is non-critical
 			this.notify(this.options.install, localize('serverStopError', 'Error stopping Notebook Server: {0}', utils.getErrorMessage(error)));
 		} finally {
-			this._isStarted = false;
-			this.utils.ensureProcessEnded(this.childProcess);
-			this.handleConnectionClosed();
+			if (this._attachedEditorCount === 0) {
```

Comment:
Why do you only do this if there's no attached editors? You've already ran the stop command and deleted the folder so I'm not sure why you'd need to check this here (it should never not be 0 here anyways I would think)